sky/skylet/providers/command_runner.py

```diff
@@ -143,7 +143,9 @@ def run_init(self, *, as_head: bool, file_mounts: Dict[str, str],
                 docker_login_config.password,
                 docker_login_config.server,
             ))
-            specific_image = f'{docker_login_config.server}/{specific_image}'
+            server_prefix = f'{docker_login_config.server}/'
+            if not specific_image.startswith(server_prefix):
+                specific_image = f'{server_prefix}{specific_image}'
```

Comment:
Should we handle this here or the `_with_docker_login_config` in resources? 
Pro for latter: we can assume the information provided to the node_provider or new provisioner always correct.